.github/workflows/ci.yml

```diff
@@ -0,0 +1,71 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - "master"
+      - "maintenance/.+"
+  pull_request:
+    branches:
+      - "master"
+      - "maintenance/.+"
+      # temporary; remove before merging #29
+      - "packaging"
+  schedule:
+    # Run a cron job once daily
+    - cron: "0 0 * * *"
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        python-version: [3.6]
+    env:
+      PYVER: ${{ matrix.python-version }}
+      PACKAGE: teachopencadd # Replace with the name of your package
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: goanpeca/setup-miniconda@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+          activate-environment: teachopencadd
+          channel-priority: true
+          environment-file: environment.yml
+          auto-activate-base: false
+
+      - name: Additional info about the build
+        shell: bash
+        run: |
+          uname -a
+          df -h
+          ulimit -a
+
+      - name: Environment Information
+        shell: bash -l {0}
+        run: |
+          conda info --all
+          conda list
+
+      - name: Run tests
+        shell: bash -l {0}
+        run: |
+          PYTEST_ARGS=" --nbval"
+          pytest $PYTEST_ARGS $PACKAGE
+
+  treebeard:
+    runs-on: ubuntu-latest
+    name: Run treebeard
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - uses: treebeardtech/treebeard@master
+        with:
+          api-key: ${{ secrets.TREEBEARD_API_KEY }}
+          docker-username: "jaimergp"
+          docker-password: "${{ secrets.DOCKER_PASSWORD }}"
```

Comment:
A question so that I better understand what is happening: 
Where does the `secrets` variable live, somewhere on your computer?

Reply:
Nope, this a special namespace in GitHub Actions. Secrets are defined in the admin panel of the repo (or organization) and exposed in that special namespace. I don't know if you have access, but feel free to explore in Settings> Secrets for any repo of your choosing. 

[More info](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets)

Reply:
Cool, thanks for clarifying! I have access :)